app/extensions/flask_sqlalchemy/__init__.py

```diff
@@ -45,9 +45,9 @@ class SQLAlchemy(BaseSQLAlchemy):
     """
 
     def __init__(self, *args, **kwargs):
-        if 'session_options' not in kwargs:
-            kwargs['session_options'] = {}
-        kwargs['session_options']['autocommit'] = True
+        # if 'session_options' not in kwargs:
+        #     kwargs['session_options'] = {}
+        # kwargs['session_options']['autocommit'] = True
```

Comment:
We need to sort this out one way or another since it is quite a breaking change.

Reply:
Yeah I had a lot of issues with autocommit, namely with sqlalchemy complaining that a transaction already was started when `with db.session.begin():` was called. In authlib, the complaints were about no transaction has started. In my own project I turned it off, because I had my own transaction system, and also I was using postgresql.